pkg/storj/pieceid.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storj
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/hex"
+
+	"github.com/zeebo/errs"
+)
+
+// ErrPieceID is used when something goes wrong with a piece ID
+var ErrPieceID = errs.Class("piece ID error")
+
+// PieceID2 is the unique identifier for pieces
+type PieceID2 [32]byte
+
+// NewPieceID creates a piece ID
+func NewPieceID() PieceID2 {
+	var id PieceID2
+
+	_, err := rand.Read(id[:])
+	if err != nil {
+		panic(err)
+	}
+
+	return id
+}
+
+// PieceIDFromString decodes a hex encoded piece ID string
+func PieceIDFromString(s string) (PieceID2, error) {
+	idBytes, err := hex.DecodeString(s)
```

Comment:
How about using using Base32 encoding instead of Hex/Base16? Base32 is still single case, but it generates shorter encoded strings and uses more letter than numbers. I find it weird to have the piece file names with so many numbers.

Reply:
This is not the encoding that filestoring will use. This is just for printing.

Reply:
Nevertheless, when I am debugging my eyes will feel better with this:
```
JEQGY2LLMUQEEYLTMUZTEIDCMV2HIZLSEB2GQYLOEBUGK6BB
```
than this:
```
49206C696B652042617365333220626574746572207468616E2068657821
```

Reply:
fixed